creme/datasets/synth/led.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+
+from .. import base
+from creme.utils.skmultiflow_utils import check_random_state
+
+
+class LED(base.SyntheticDataset):
+    """ LED stream generator.
+
+    This data source originates from the CART book [^1]. An implementation
+    in C was donated to the UCI [^2] machine learning repository by David Aha.
+    The goal is to predict the digit displayed on a seven-segment LED display,
+    where each attribute has a 10% chance of being inverted. It has an optimal
+    Bayes classification rate of 74%. The particular configuration of the
+    generator used for experiments (LED) produces 24 binary attributes,
+    17 of which are irrelevant.
+
+    Parameters
+    ----------
+    seed
+        If int, `seed` is used to seed the random number generator;
+        If RandomState instance, `seed` is the random number generator;
+        If None, the random number generator is the `RandomState` instance used
+        by `np.random`.
+    noise_percentage
+        The probability that noise will happen in the generation. At each
+        new sample generated, a random number is generated, and if it is equal
+        or less than the noise_percentage, the led value  will be switched
+    irrelevant_features
+        Adds 17 non-relevant attributes to the stream.
+
+    Notes
+    -----
+    An instance is generated based on the parameters passed. If `has_noise`
+    is set then the total number of attributes will be 24, otherwise there will
+    be 7 attributes.
+
+    References
```

Comment:
Normally the references go after the example. This won't affect the docs in any case.

Reply:
I was following the Numpy docstring format which places the example at the end.

https://numpydoc.readthedocs.io/en/latest/format.html

I don't mind changing the location. It might be worth documenting the docstring sections order for consistency.

Reply:
I have reordered the sections, now `Notes` and `References` are placed after the examples. This is since arguably the examples are more relevant to the user.